src/ExternalApi/Isite/Mapper/RowMapper.php

```
@@ -0,0 +1,42 @@
+<?php
+declare(strict_types = 1);
+
+namespace App\ExternalApi\Isite\Mapper;
+
+use App\ExternalApi\Isite\Domain\Row;
+use SimpleXMLElement;
+
+class RowMapper extends Mapper
+{
+    public function getDomainModel(SimpleXMLElement $isiteObject): Row
+    {
+        return new Row(
+            $this->extractBlocks($isiteObject, 'primary'),
+            $this->extractBlocks($isiteObject, 'secondary')
+        );
+    }
+
+    private function extractBlocks(SimpleXMLElement $isiteObject, string $type): ?array
+    {
+        //check if module is in the data
+        if (empty($isiteObject->{$type})) {
+            return [];
+        }
+
+        $blocks = $isiteObject->{$type};
+        $name = $type . '-blocks';
+        if (empty($blocks[0]->{$name}->result)) {
+            return null; // Content blocks have not been fetched
```

Do we want to throw an error rather than return null?

---

@BjornTwachtmann any view?

---

Probably make it throw if this shouldn't happen